portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtServiceImpl.java

```diff
@@ -66,6 +67,22 @@
     private NotificationPasswordRecoveryManager notificationPasswordRecoveryManager;
     private SecurityQuestionPasswordRecoveryManager securityQuestionPasswordRecoveryManager;
     private NotificationUsernameRecoveryManager notificationUsernameRecoveryManager;
+    private PasscodeGenerator passcodeGenerator = new DefaultPasscodeGenerator();
+    private static final int MAX_LENGTH = 6;
+
+    @Reference(
+            name = "PasscodeGeneratorService",
+            service = PasscodeGenerator.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetPasscodeGeneratorService")
+    protected void setPasscodeGeneratorService(PasscodeGenerator passcodeGenerator) {
+        this.passcodeGenerator = passcodeGenerator;
+    }
+
+    protected void unsetPasscodeGeneratorService(PasscodeGenerator passcodeGenerator) {
+        this.passcodeGenerator = null;
```

Comment:
passcodeGenerator = new DefaultPasscodeGenerator();
We need to reset to default class when we unset the contribution from extension.